scipy/spatial/_hausdorff.py

```diff
@@ -0,0 +1,61 @@
+"""
+Directed Hausdorff Code
+
+.. versionadded:: 0.19.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy, Richard Gowers, and Max Linke, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+# adapted from scipy's cython version
+#pythran export directed_hausdorff(float64[:,:], float64[:,:], int or None)
+#pythran export directed_hausdorff(float64[:,:], float64[:,:])
+
+import numpy as np
+import random
+
+def directed_hausdorff(ar1, ar2, seed=None):
+
+    if seed is None:
+        seed = 0
+
+    N1, data_dims = ar1.shape
+    N2 = ar2.shape[0]
+    i_store = j_store = i_ret = j_ret = 0
+
+    # shuffling the points in each array generally increases the likelihood of
+    # an advantageous break in the inner search loop and never decreases the
+    # performance of the algorithm
+    random.seed(seed)
```

Comment:
Is this identical to `rng = np.random.RandomState(seed)` used in the current Cython implementation?

Even the slightest change in random shuffling behavior may affect that sensitive test failure you are seeing on OSX.